src/main/java/vimification/taskui/TaskListPanel.java

```diff
@@ -28,21 +28,74 @@ public TaskListPanel(ObservableList<Task> taskList) {
         taskListView.setCellFactory(listView -> new TaskListViewCell());
     }
 
-    /**
-     * Custom {@code ListCell} that displays the graphics of a {@code Task} using a
-     * {@code TaskCard}.
-     */
-    class TaskListViewCell extends ListCell<Task> {
-        @Override
-        protected void updateItem(Task task, boolean empty) {
-            super.updateItem(task, empty);
-
-            if (empty || task == null) {
-                setGraphic(null);
-                setText(null);
-            } else {
-                setGraphic(new TaskCard(task, getIndex() + 1).getRoot());
-            }
+    public void setMainScreen(MainScreen parent) {
+        this.mainScreen = parent;
+    }
+
+    public void requestFocus() {
+        this.taskListView.requestFocus();
+    }
+
+    @FXML
+    private void initialize() {
+        // TODO: Implement Visual Mode
+        // taskListView.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
+        this.getRoot().setFocusTraversable(true);
+        taskListView.setFocusTraversable(true);
+        Platform.runLater(() -> {
+            // Hackish way of requesting focus
+            // after every listItem has been
+            // loaded
+            taskListView.requestFocus();
+            taskListView.getSelectionModel().selectFirst();
+        });
+    }
+
+    @FXML
+    private void handleKeyPressed(KeyEvent event) {
+        // System.out.println("You are navigating in TaskListPanel");
+        switch (event.getText()) {
+        case "d":
+            System.out.println("You've deleted a task!");
+            break;
+        case "l":
+            Task selectedTask = taskListView.getSelectionModel().getSelectedItem();
+            mainScreen.loadDetailedTaskComponent(selectedTask);
+            break;
+        case "j":
+            System.out.println("You've moved down");
+            navigateToNextCell();
+            break;
+        case "k":
+            System.out.println("You've moved up");
+            navigateToPrevCell();
+            break;
+        }
```

Comment:
Refactor to use KeyCode in the future if possible.

Reply:
I agree, this is just temporary code for now. Will add it to my TODO